File: seo-product/frontend/src/types/keyword.ts
export interface KeywordGap {
  id: string;
  project_id: string;
  keyword: string;
  volume: number;
  kd: number;
  intent: string | null;
  competitor_domains: string[];
  cpc: number | null;
  difficulty: string | null;
  category: string | null;
  city: string | null;
  is_long_tail: boolean | null;
  tier: string | null;
  word_count: number | null;
  fetched_at: string;
}

export interface KeywordGapList {
  items: KeywordGap[];
  total_count: number;
  page: number;
  page_size: number;
}

export interface LongTailKeyword {
  id: string;
  keyword_gap_id: string;
  keyword: string;
  volume: number;
  kd: number;
  intent: string | null;
}

export interface SelectedKeyword {
  id: string;
  project_id: string;
  keyword: string;
  volume: number;
  kd: number;
  intent: string | null;
  long_tail_keywords: LongTailKeyword[];
  status: string;
  created_at: string;
}

export interface KeywordFilters {
  kd_min?: number;
  kd_max?: number;
  vol_min?: number;
  vol_max?: number;
  intent?: string;
  search?: string;
  category?: string;
  sort_by?: string;
  sort_dir?: string;
  page?: number;
  page_size?: number;
}

// Fixed filter tabs (KD-based, always visible)
export const FIXED_TABS = [
  { label: "All", value: "" },
  { label: "Easy Wins (KD 0-10)", value: "__easy_wins" },
  { label: "Quick Wins (KD 0-15)", value: "__quick_wins" },
  { label: "Long-Tail Gold", value: "__long_tail" },
] as const;

// Hardcoded category tabs per niche
export const NICHE_CATEGORIES: Record<string, string[]> = {
  // Apartment / rental / renter niches
  apartment: [
    "Renter Education",
    "Price & Budget",
    "pet friendly",
    "Bedroom Type",
    "Property Type",
    "Near Me",
    "Student & Campus",
    "Luxury",
    "Furnished & Short-Term",
    "General",
  ],
  // Plumbing niches
  plumbing: [
    "Emergency Repairs",
    "Drain & Sewer",
    "Water Heater",
    "Fixture Installation",
    "Bathroom Plumbing",
    "Kitchen Plumbing",
    "Commercial Plumbing",
    "Pipe & Leak Repair",
    "Cost & Pricing",
    "General",
  ],
  // Real estate niches
  "real estate": [
    "Buying Guide",
    "Selling Guide",
    "Mortgage & Finance",
    "First-Time Buyer",
    "Investment Property",
    "Market Trends",
    "Home Inspection",
    "Neighborhood Guide",
    "General",
  ],
  // Default fallback
  default: ["General"],
};

export function getCategoryTabs(niche: string | null): string[] {
  if (!niche) return NICHE_CATEGORIES.default;
  const lower = niche.toLowerCase();
  for (const [key, cats] of Object.entries(NICHE_CATEGORIES)) {
    if (key === "default") continue;
    if (lower.includes(key)) return cats;
  }
  return NICHE_CATEGORIES.default;
}

File: seo-product/backend/app/schemas/keyword.py
from pydantic import BaseModel
from datetime import datetime


class KeywordGapResponse(BaseModel):
    id: str
    project_id: str
    keyword: str
    volume: int
    kd: int
    intent: str | None
    competitor_domains: list[str]
    cpc: float | None = 0
    difficulty: str | None = None
    category: str | None = None
    city: str | None = None
    is_long_tail: bool | None = False
    tier: str | None = None
    word_count: int | None = 0
    fetched_at: datetime


class KeywordGapListResponse(BaseModel):
    items: list[KeywordGapResponse]
    total_count: int
    page: int
    page_size: int


class LongTailResponse(BaseModel):
    id: str
    keyword_gap_id: str
    keyword: str
    volume: int
    kd: int
    intent: str | None


class KeywordImportItem(BaseModel):
    keyword: str
    volume: int = 0
    kd: int = 0
    cpc: float = 0
    intent: str | None = None
    category: str | None = None
    city: str | None = None


class KeywordImportRequest(BaseModel):
    keywords: list[KeywordImportItem]
    replace: bool = False  # If true, replace all existing keywords


class SelectedKeywordCreate(BaseModel):
    keyword: str
    volume: int = 0
    kd: int = 0
    intent: str | None = None
    long_tail_keywords: list[dict] = []


class SelectedKeywordResponse(BaseModel):
    id: str
    project_id: str
    keyword: str
    volume: int
    kd: int
    intent: str | None
    long_tail_keywords: list[dict]
    status: str
    created_at: datetime
